File: src/main/java/com/github/sorusclient/client/module/impl/fullbright/v1_8_9/FullBrightTransformer.java
package com.github.sorusclient.client.module.impl.fullbright.v1_8_9;

import com.github.glassmc.loader.GlassLoader;
import com.github.glassmc.loader.Listener;
import com.github.glassmc.loader.loader.ITransformer;
import com.github.glassmc.loader.util.Identifier;
import com.github.sorusclient.client.transform.Applier;
import com.github.sorusclient.client.transform.Transformer;
import org.objectweb.asm.ClassReader;
import org.objectweb.asm.ClassWriter;
import org.objectweb.asm.Opcodes;
import org.objectweb.asm.tree.*;

public class FullBrightTransformer extends Transformer implements Listener {

    @Override
    public void run() {
        GlassLoader.getInstance().registerTransformer(FullBrightTransformer.class);
    }

    public FullBrightTransformer() {
        this.register("v1_8_9/net/minecraft/client/render/GameRenderer", this::transformGameRenderer);
    }

    private void transformGameRenderer(ClassNode classNode) {
        Identifier updateLightmap = Identifier.parse("v1_8_9/net/minecraft/client/render/GameRenderer#updateLightmap(F)V");

        /*this.findMethod(classNode, updateLightmap)
                .apply(methodNode -> this.findVarReferences(methodNode, 17, VarReferenceType.STORE)
                        .apply(new Applier.InsertBefore<>(methodNode, this.getHook("modifyGamma"))));*/
        for (MethodNode methodNode : classNode.methods) {
            if (methodNode.name.equals(updateLightmap.getMethodName()) && methodNode.desc.equals(updateLightmap.getMethodDesc())) {
                for (AbstractInsnNode node : methodNode.instructions.toArray()) {
                    if (node.getOpcode() == Opcodes.FSTORE && ((VarInsnNode) node).var == 17 && node.getPrevious() instanceof FieldInsnNode) {
                        methodNode.instructions.insertBefore(node, new MethodInsnNode(Opcodes.INVOKESTATIC, FullBrightHook.class.getName().replace(".", "/"), "modifyGamma", "(F)F"));
                    }
                }
            }
        }
    }

}

File: src/main/kotlin/com/github/sorusclient/client/module/impl/fullbright/v1_8_9/FullBrightTransformer.kt
package com.github.sorusclient.client.module.impl.fullbright.v1_8_9

import com.github.glassmc.loader.GlassLoader
import com.github.glassmc.loader.Listener
import com.github.glassmc.loader.util.Identifier
import com.github.sorusclient.client.transform.Transformer
import org.objectweb.asm.Opcodes
import org.objectweb.asm.tree.ClassNode
import org.objectweb.asm.tree.FieldInsnNode
import org.objectweb.asm.tree.MethodInsnNode
import org.objectweb.asm.tree.VarInsnNode

class FullBrightTransformer : Transformer(), Listener {
    override fun run() {
        GlassLoader.getInstance().registerTransformer(FullBrightTransformer::class.java)
    }

    init {
        register("v1_8_9/net/minecraft/client/render/GameRenderer") { classNode: ClassNode ->
            transformGameRenderer(
                classNode
            )
        }
    }

    private fun transformGameRenderer(classNode: ClassNode) {
        val updateLightmap = Identifier.parse("v1_8_9/net/minecraft/client/render/GameRenderer#updateLightmap(F)V")

        for (methodNode in classNode.methods) {
            if (methodNode.name == updateLightmap.methodName && methodNode.desc == updateLightmap.methodDesc) {
                for (node in methodNode.instructions.toArray()) {
                    if (node.opcode == Opcodes.FSTORE && (node as VarInsnNode).`var` == 17 && node.getPrevious() is FieldInsnNode) {
                        methodNode.instructions.insertBefore(
                            node,
                            MethodInsnNode(
                                Opcodes.INVOKESTATIC,
                                FullBrightHook::class.java.name.replace(".", "/"),
                                "modifyGamma",
                                "(F)F"
                            )
                        )
                    }
                }
            }
        }
    }
}